users/arxiv/users/legacy/cookies.py

```diff
@@ -53,10 +51,13 @@ def unpack(cookie: str) -> Tuple[str, str, str, datetime, datetime, str]:
     capabilities = parts[4]
     try:
         expected = pack(session_id, user_id, ip, issued_at, capabilities)
-        assert expected == cookie
-    except (TypeError, AssertionError) as e:
-        raise InvalidCookie('Invalid session cookie; forged?') from e
-    return session_id, user_id, ip, issued_at, expires_at, capabilities
+    except Exception as e:
+        raise InvalidCookie('Invalid session cookie; problem while repacking') from e
```

Comment:
Is this invalid after repacking error common? I'm curious since it appears to be repacking the cookie using existing verified data.

Reply:
Yes, this repack is a check of an incoming login request. The cookie is repacked which signs it and it is expected to match.